src/conn/conn_reconfig.c

```diff
@@ -94,64 +90,61 @@ __wt_conn_compat_config(WT_SESSION_IMPL *session, const char **cfg, bool reconfi
      * now. This is on an open and we're checking the two against each other. We'll check against
      * what was saved on a restart later.
      */
-    if (!reconfig && max_major != WT_CONN_COMPAT_NONE &&
-      (max_major < rel_major || (max_major == rel_major && max_minor < rel_minor)))
+    if (!reconfig && __wt_version_defined(max_compat) && __wt_version_lt(max_compat, new_compat))
         WT_RET_MSG(session, ENOTSUP,
           WT_COMPAT_MSG_PREFIX "required max of %" PRIu16 ".%" PRIu16
                                "cannot be smaller than compatibility release %" PRIu16 ".%" PRIu16,
-          max_major, max_minor, rel_major, rel_minor);
+          max_compat.major, max_compat.minor, new_compat.major, new_compat.minor);
 
     /*
      * The minimum required must be less than or equal to the compatibility release we're using now.
      * This is on an open and we're checking the two against each other. We'll check against what
      * was saved on a restart later.
      */
-    if (!reconfig && min_major != WT_CONN_COMPAT_NONE &&
-      (min_major > rel_major || (min_major == rel_major && min_minor > rel_minor)))
+    if (!reconfig && __wt_version_defined(min_compat) && __wt_version_gt(min_compat, new_compat))
         WT_RET_MSG(session, ENOTSUP,
           WT_COMPAT_MSG_PREFIX "required min of %" PRIu16 ".%" PRIu16
                                "cannot be larger than compatibility release %" PRIu16 ".%" PRIu16,
-          min_major, min_minor, rel_major, rel_minor);
+          min_compat.major, min_compat.minor, new_compat.major, new_compat.minor);
 
     /*
      * On a reconfigure, check the new release version against any required maximum version set on
      * open.
      */
-    if (reconfig && conn->req_max_major != WT_CONN_COMPAT_NONE &&
-      (conn->req_max_major < rel_major ||
-        (conn->req_max_major == rel_major && conn->req_max_minor < rel_minor)))
+    if (reconfig && __wt_version_defined(conn->compat_req_max) &&
+      __wt_version_lt(conn->compat_req_max, new_compat))
         WT_RET_MSG(session, ENOTSUP,
           WT_COMPAT_MSG_PREFIX "required max of %" PRIu16 ".%" PRIu16
                                "cannot be smaller than requested compatibility release %" PRIu16
                                ".%" PRIu16,
-          conn->req_max_major, conn->req_max_minor, rel_major, rel_minor);
+          conn->compat_req_max.major, conn->compat_req_max.minor, new_compat.major,
+          new_compat.minor);
 
     /*
      * On a reconfigure, check the new release version against any required minimum version set on
      * open.
      */
-    if (reconfig && conn->req_min_major != WT_CONN_COMPAT_NONE &&
-      (conn->req_min_major > rel_major ||
-        (conn->req_min_major == rel_major && conn->req_min_minor > rel_minor)))
+    if (reconfig && __wt_version_defined(conn->compat_req_min) &&
+      __wt_version_gt(conn->compat_req_min, new_compat))
```

Comment:
Yep, you're right we're now checking `minor` as well 👍 . 
I've made this change because whenever `major` is set in the code `minor` is set as well. We also require minor version info for all of our `WT_LOG_V*_VERSION` checks (see the original definitions in `src/include/log.h`).

Based on this I think there was an assumption hat when `major` is undefined `minor` is undefined also, and there was no need to check for this. `__wt_version_defined` is a bit more precise but the end result is the same.

Reply:
Great, thanks for explaining that.